node/src/cli.rs

```diff
@@ -124,7 +138,7 @@ where
 // handles ctrl-c
 pub struct Exit;
 impl IntoExit for Exit {
-    type Exit = future::MapErr<oneshot::Receiver<()>, fn(oneshot::Canceled) -> ()>;
+    type Exit = Map<oneshot::Receiver<()>, fn(Result<(), oneshot::Canceled>) -> ()>;
```

Comment:
Just a heads-up - `clippy::all` is now required by CI, and this typedef will raise a lint warning.
```suggestion
    type Fut = oneshot::Receiver<()>;
    type Func = fn(Result<(), oneshot::Canceled>) -> ();
    type Exit = Map<Fut, Func>;
```

Reply:
I added an `#[allow(..)]`